src/rgw/rgw_rados.cc

```diff
@@ -8097,6 +8107,13 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
   if (ret < 0) {
     return ret;
   }
+  if (rgw_s3_is_encrypted(src_attrs)) {
```

Comment:
i think this could just be `if (src_atts.count(RGW_ATTR_CRYPT_MODE))` and we wouldn't need the extra member function